static/js/metrics-explorer.js

```diff
@@ -1681,31 +1695,31 @@ function mergeGraphs(chartType, panelId = -1) {
         datasets: [],
     };
     var graphNames = [];
-    let colorIndex = 0;
 
     // Loop through chartDataCollection to merge datasets
     for (var queryName in chartDataCollection) {
         if (Object.prototype.hasOwnProperty.call(chartDataCollection, queryName)) {
             // Merge datasets for the current query
             var datasets = chartDataCollection[queryName].datasets;
             graphNames.push(`${datasets[0].label}`);
-
-            datasets.forEach(function (dataset, datasetIndex) {
+            datasets.forEach(function (dataset) {
                 // Calculate color for the dataset
-                let datasetColor = classic[(colorIndex + datasetIndex) % classic.length];
+                let datasetColor = colorPalette.Classic[colorIndex % colorPalette.Classic.length];
```

Comment:
Looks good. Just one thing: instead of using "classic," you could use the selected theme.

`let datasetColor = colorPalette[selectedTheme][(colorIndex) % colorPalette[selectedTheme].length];`